fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -700,7 +702,14 @@ private Query getFilterQuery(Set<ApiFilter> filters) {
                         perPage
                 );
                 hits = hitDocs.scoreDocs;
-                documentCount = hitDocs.totalHits;
+                if (hitDocs.totalHits > Integer.MAX_VALUE) {
+                    String message = String.format(TOO_MANY_DOCUMENTS, hitDocs.totalHits);
+                    IllegalStateException illegalStateException = new IllegalStateException(message);
+                    LOG.error(illegalStateException.getMessage(), illegalStateException);
+                    throw illegalStateException;
```

Comment:
Why not use the same exception for when `perPage` exceeds `maxResults`: a `RowLimitReachedException`? Seems to me like this is a similar kind of problem (we're getting back more results than we can handle). For reference: https://github.com/yahoo/fili/pull/824/files#diff-0e7dd2ec8a518d234151c865cd2b6dacR764